eox_nelp/openedx_filters/tests/xapi/tests_filters.py

```diff
@@ -263,41 +278,49 @@ def test_update_description(self, get_course_mock):
             returned_activity.definition.description,
         )
 
+    @data(constants.XAPI_ACTIVITY_MODULE, constants.XAPI_ACTIVITY_QUESTION)
     @patch("eox_nelp.openedx_filters.xapi.filters.get_course_from_id")
-    def test_empty_label(self, get_course_mock):
+    def test_empty_label(self, definition_type, get_course_mock):
         """ Test case when the item markdown doesn't contain a label.
 
         Expected behavior:
             - Definition description is an empty dict.
         """
         get_course_mock.return_value = self.course
+        self.set_activity(definition_type)
 
         returned_activity = self.filter.run_filter(transformer=self.transformer, result=self.activity)["result"]
 
         self.assertEqual({}, returned_activity.definition.description)
 
+    @data(constants.XAPI_ACTIVITY_MODULE, constants.XAPI_ACTIVITY_QUESTION)
     @patch("eox_nelp.openedx_filters.xapi.filters.get_course_from_id")
-    def test_default_language(self, get_course_mock):
+    def test_default_language(self, definition_type, get_course_mock):
         """ Test case when the course has no language or is not valid.
 
         Expected behavior:
             - Definition name has the default key language.
         """
         get_course_mock.return_value = {}
+        self.set_activity(definition_type)
 
         returned_activity = self.filter.run_filter(transformer=self.transformer, result=self.activity)["result"]
 
         self.assertEqual([DEFAULT_LANGUAGE], list(returned_activity.definition.name.keys()))
 
     def test_invalid_event(self):
```

Comment:
For this test I have some doubts about why if in the pipe you remove the normal exist of the event. Why this test seem without problem?.

Reply:
the criteria of the event changed this has an invalid definition type that's why this works, to be honest and not sure if I understood your question